cmd/shell.go

```diff
@@ -260,10 +260,47 @@ type CAConfig struct {
 }
 
 // PAConfig specifies how a policy authority should connect to its
-// database, and what policies it should enforce.
+// database, what policies it should enforce, and what challenges
+// it should offer.
 type PAConfig struct {
 	DBConnect              string
 	EnforcePolicyWhitelist bool
+	Challenges             map[string]bool
+}
+
+// UnmarshalJSON is really actually vanilla, but with some validity checks and
```

Comment:
I think my objections is more to the defaulting than to the validation. If I
were debugging a situation where the provided challenge set didn't appear to
match what was in the JSON config, I would be surprised to find that
UnmarshalJSON was setting its output to something that didn't match its input.

For validation there's a question of where, in general, we want to put
validation of JSON input beyond the purely syntactic. I tend to lean towards
doing the validation explicitly and close to where it's used, i.e. in the VA.
That way an edit in cmd/ is less likely to accidentally disable a validation
that the VA was depending on, e.g. by changing a type.

Another way of putting it: We can't treat configs as semantically valid by virtue of  having been parsed, and so it's confusing to try to do so in a subset of cases.